src/porepy/numerics/ad/equation_system.py

```diff
@@ -1129,6 +1144,8 @@ def set_equation(
             self._equation_image_size_info.update({name: equations_per_grid_entity})
             # Store the equation itself.
             self._equations.update({name: equation})
+            # Store the linear/nonlinear flag.
```

Comment:
Can't we do this at the top of the method to avoid the double registration (here and towards the end)?